### ISSUE TITLE
Support unknown reader tags while reading EDN

### ISSUE THREAD

**Issue by thenonameguy** [borkdude/jet#58]
There should be an easy way to pass through tagged literals that are unknown through jet.
Recommendation:
Augment https://github.com/borkdude/jet/blob/4c93fde5e5766fe9b95e46b1e343db1686354583/src/jet/formats.clj#L33
to support `:default`, with a CLI flag `--edn-read-default=sci-expr`.

**Comment by borkdude** [comment]
Interesting idea. Would it make sense to also support other data readers by passing a map? Maybe we could just pass in additional args to tools.reader with an option `--tools-reader-opts` which acceps a map with `:readers` and  `:default` which will be evaluated by sci?

**Comment by thenonameguy** [comment]
Yes, that could also work. Alternatively I would be okay with a simple boolean flag to enable `tagged-literal` so that passthrough works: https://insideclojure.org/2018/06/21/tagged-literal/

**Comment by borkdude** [comment]
The above combined:

Allow --edn-opts with:

- `:readers` option with accepts map of symbol -> function
- `:default` which you can pass `tagged-literal`. So a combination of the above.

The entire map will be evaluated by sci. For this to work I'd have to add tagged-literal to sci first.

Example call:

``` clojure
echo '[#foo{:a 1} #bar {:a 2}]' | jet --from edn --edn-opts '{:readers {foo (fn [x] [:foo x])} :default tagged-literal}'
```

Output:

```
[[:foo {:a 1}] #bar {:a 2}]
```

**Comment by borkdude** [comment]
Fixed in v0.0.12 by adding the `--edn-reader-opts` option.
